bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/i18n/LocaleProvider.java

```diff
@@ -25,4 +28,10 @@
      */
     Locale getLocale();
 
+    /**
+     * Provides access to the location of the installation
+     *
+     * @return location of the current installation or null if the location is not set
+     */
+    PointType getLocation();
```

Comment:
>  if I need to translate some stuff I would like an I18nProvider.

No, definitely not - you would like a TranslationProvider. "i18n" is a wide term, which covers locale (and possibly location). So having it in place, but ONLY providing localised (!) texts, is weird.

So the decision had already been taken a long time ago when the I18nProvider has been introduced: The "set of stuff that should be combined" is the functionality around i18n - that's why it is most logical to me to merge the implementations.

Another consideration has to be what is "configured together", i.e. what should have an entry in ConfigAdmin and also appear as a single service e.g. in the Paper UI Configuration->Services section. Here as well, I think it makes sense to have those settings together in an "i18n" section. Splitting it in smaller pieces will only make it harder to find stuff.